modules/build/src/main/scala/scala/build/errors/Severity.scala

```diff
@@ -0,0 +1,8 @@
+package scala.build.errors
+
+sealed trait Severity
```

Comment:
I'm not sure we need that class (and the additional field in `BuildException`). `BuildException`s that should be considered as warnings should be reported via a different channel than the left side of a `Either[BuildException, …]`. As if there are only warnings, such an either should be `Right(…)`, and whatever was happening can go on.

Reply:
That also makes invalid states "representable", as one could return a `Left(ex)` with `ex` having only warning severity, which would interrupt the processing while it shouldn't.

Reply:
I kept this because I extracted diagnostics logic in a separate class `Diagnostic` that is not derived from `BuildException`